indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/StructuredData.java

```diff
@@ -354,16 +359,21 @@ private NamedProperty getProperty(String propertyName, Collection<Object> values
       if (nonNullValues.isEmpty()) {
         return null;
       }
-      if (!isRepeated) {
-        return propertyBuilder.getNamedProperty(
-            propertyName, Collections.singletonList(valueConverter.convert(nonNullValues.get(0))));
-      } else {
-        return propertyBuilder.getNamedProperty(
-            propertyName,
-            nonNullValues
-                .stream()
-                .map(v -> valueConverter.convert(v))
-                .collect(Collectors.toList()));
+      try {
+        if (!isRepeated) {
+          return propertyBuilder.getNamedProperty(
+              propertyName, Collections.singletonList(valueConverter.convert(nonNullValues.get(0))));
+        } else {
+          return propertyBuilder.getNamedProperty(
+              propertyName,
+              nonNullValues
+                  .stream()
+                  .map(v -> valueConverter.convert(v))
+                  .collect(Collectors.toList()));
+        }
+      } catch (IllegalArgumentException e) {
```

Comment:
In case of multi value / repeated fields, an invalid value would result into skipping all values. Do we want to ignore just failing value or drop entire property?

Reply:
Yes, I see, it's probably better to skip only the bad value in the list. So, we probably cannot call valueConverter::convert directly. We would need to introduce a private method that does the conversion and swallows the exception. I will take another attempt at that.

Reply:
I have pushed this fix for multi value properties. Only the bad values are dropped, others are kept. If it reduces the list to empty, it returns null so the property is not set.